File: C++/561. Array Partition I.cpp
// Time:  O(nlog(n))
// Space: O(1)
class Solution {
public:
    int arrayPairSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int ans = 0;
        for(unsigned i=0;i<nums.size();i+=2)
            ans += nums[i];
        return ans;
    }
};

File: Python/561. Array Partition I.py
# Time:  O(nlog(n))
# Space: O(1)
class Solution:
    def arrayPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        ans = 0
        for i in range(0, len(nums), 2):
            ans += nums[i]
        return ans